smartsim/_core/_cli/build.py

```diff
@@ -30,104 +30,38 @@
 import typing as t
 from pathlib import Path
 
-import pkg_resources
 from tabulate import tabulate
 
-from smartsim._core._cli.utils import (color_bool, pip_install,
-                                       SMART_LOGGER_FORMAT)
+from smartsim._core._cli import verify
+from smartsim._core._cli.utils import (
+    SMART_LOGGER_FORMAT,
+    color_bool,
+    pip_install,
+    pip_uninstall,
+)
 from smartsim._core._install import builder
 from smartsim._core._install.buildenv import (
     BuildEnv,
+    DbEngine,
     SetupError,
     Version_,
+    VersionConflictError,
     Versioner,
-    DbEngine,
 )
 from smartsim._core._install.builder import BuildError
 from smartsim._core.config import CONFIG
 from smartsim._core.utils.helpers import installed_redisai_backends
 from smartsim.error import SSConfigError
 from smartsim.log import get_logger
 
-SMART_LOGGER_FORMAT = "[%(name)s] %(levelname)s %(message)s"
 logger = get_logger("Smart", fmt=SMART_LOGGER_FORMAT)
 
 # NOTE: all smartsim modules need full paths as the smart cli
 #       may be installed into a different directory.
 
```

Comment:
Would you mind putting a brief description for this module? It's definitely one of the weird bits that whenever we ask a dev to touch it's not immediately understandable